nodestream_plugin_neptune/database_connector.py

```diff
@@ -1,17 +1,88 @@
-from nodestream.databases import TypeRetriever, DatabaseConnector
-from nodestream.databases.query_executor import QueryExecutor
+from nodestream.databases.copy import TypeRetriever
+from nodestream.databases.database_connector import QueryExecutor, DatabaseConnector
+from .ingest_query_builder import NeptuneDBIngestQueryBuilder
 
-from .query_executor import NeptuneQueryExecutor
-from .type_retriever import NeptuneTypeRetriever
 
+class NeptuneDatabaseConnector(DatabaseConnector, alias="neptune_db"):
```

Comment:
How do you feel about keeping this one called `neptune`? 

I would have to change this pr, which would mean that people would have to use `nodestream new my_project --database neptune_db` which feels a bit less ergonomic than `nodestrea new my_project --database neptune`

Reply:
I agree that the conciseness of `neptune` is nice. Do you have thoughts on using `neptune_analytics` as the alias for Neptune Analytics? Do you have any thoughts on having a single `neptune` database connector which has some parameter which switches it between neptune db and neptune analytics?

Reply:
I do like the idea of using a consolidating into one connector. Couple ways to implement this that I can think of. Chad and I spoke and we've kind of come to like this approach. Here's how that could look. 

Analytics:

```yaml
targets:
  my-db:
   database: neptune 
   mode: analytics 
   graph_id: XXXX
```

Transactional:

```yaml
targets:
  my-db:
   database: neptune 
   mode: transactional # TODO: naming of these things 
   host: XXXX
```

Thoughts on this?